api/v1alpha1/openapi.yaml

```diff
@@ -2703,16 +2721,19 @@ components:
     HookAction:
```

Comment:
@hexfusion There seems to be something weird with this type.
Previously, the generated type using the openapi generator was like this:
```
export type HookAction = (Record<string, any> | HookActionSystemdSpec | HookActionExecutableSpec);
```
whereas now it fails to generate it (there is nothing after the equal).

Could we use the same pattern as in the `RepositorySpec` for example? 
```
    RepositorySpec:
      type: object
      oneOf:
        - $ref: "#/components/schemas/GenericRepoSpec"
        - $ref: "#/components/schemas/HttpRepoSpec"
        - $ref: "#/components/schemas/SshRepoSpec
```

That means there is no top level property to identify the type though, but it's more consistent with other parts of the API and would not break the type generation for us.

Reply:
I created a bug to track this issue, as it breaks the UI type generation process.
 https://issues.redhat.com/browse/EDM-449

Reply:
The goal was to improve UX by not requiring a discriminator field. The downside is it adds complexity on implementation.

Reply:
I don't know that much about go, and IDK what are other alternatives.

Just that the [openapi-generator](https://github.com/OpenAPITools/openapi-generator) is not able to parse this information, and that only happens with this field.